### PR TITLE
fix Node net.createConnection() and Socket#connect() signatures.

### PR DESCRIPTION
**Pull request by rogierschouten**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
https://nodejs.org/api/net.html#net_net_createconnection
https://nodejs.org/api/net.html#net_new_net_socket_options
https://nodejs.org/api/net.html#net_socket_connect_options_connectlistener
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by rogierschouten** [comment]
I don't understand the failing tests - these don't seem to be caused by my commit?

File: types/node/index.d.ts [hunk 1]
```diff
@@ -2352,6 +2352,29 @@ declare module "net" {
...
+        host?: string;
+        localAddress?: string;
+        localPort?: number;
+        hints?: number;
+        family?: number;
+        lookup?: Function;
+    }
+
+    export interface IpcSocketConnectOpts {
+        path?: string;
```
Comment (Flarna) [inline types/node/index.d.ts]: According to doc the path is required for IPC connections. If I understood the code correct the presence of path property is used to decide if TCP or IPC should be used.

File: types/node/index.d.ts [hunk 2]
```diff
@@ -2352,6 +2352,29 @@ declare module "net" {
...
+    }
+
+    export interface TcpSocketConnectOpts {
+        port: number;
+        host?: string;
+        localAddress?: string;
+        localPort?: number;
+        hints?: number;
+        family?: number;
+        lookup?: Function;
```
Comment (Flarna) [inline types/node/index.d.ts]: Would be nice of the lookup property could be typed more concrete. I think it should be of same type as `dns.lookup()`, the variant using `LookupOneOptions`.

File: types/node/index.d.ts [hunk 3]
```diff
@@ -2542,12 +2567,17 @@ declare module "net" {
         prependOnceListener(event: "error", listener: (err: Error) => void): this;
         prependOnceListener(event: "listening", listener: () => void): this;
     }
+
+    export interface NetConnectOpts extends TcpSocketConnectOpts, IpcSocketConnectOpts, SocketConstructorOpts {
```
Comment (Flarna) [inline types/node/index.d.ts]: This will no longer work once property `path` in `IpcSocketConnectOpts` is mandatory.

File: types/node/index.d.ts [hunk 4]
```diff
@@ -2361,7 +2384,9 @@ declare module "net" {
         write(str: string, encoding?: string, fd?: string): boolean;
         write(data: any, encoding?: string, callback?: Function): void;
 
-        connect(port: number, host?: string, connectionListener?: Function): void;
+        connect(options: SocketConnectOpts, connectionListener?: Function): void;
```
Comment (Flarna) [inline types/node/index.d.ts]: Actually `Socket.conncet()` returns `this` not `void`.

**Review by Flarna — CHANGES_REQUESTED** [review]

**Comment by rogierschouten** [comment]
@Flarna thank you for your input, I've fixed all your comments. 

I suspect that the failing travis build is not due to my changes. I tried updating my master branch to the DT master branch and even then the tests fail. So the DT master branch doesn't seem to be stable at the moment.

**Review by Flarna — APPROVED** [review]

**Comment by Flarna** [comment]
Thanks!
I think also that the fails are not related to your changes.

**Pull request merged**